### PR TITLE
Rename BINARYEN_ASYNC_COMPILATION -> WASM_ASYNC_COMPILATION

### PR DESCRIPTION
**Pull request by sbc100**
Also, add a mechanism for supportimg renamed settings.  This enables
users in non-strict mode to use with name for the settings, but users
in strict mode can only use the modern name.

Part of #8317

### CODE REVIEW

File: tools/shared.py [hunk 1]
```diff
@@ -1186,6 +1188,8 @@ def keys(self):
       return self.attrs.keys()
 
     def __getattr__(self, attr):
+      if not self.attrs['STRICT'] and attr in SettingsManager.renamed_settings:
```
Comment (jgravelle-google) [inline tools/shared.py]: Why do we need this in getattr? It makes sense in setattr, but we should only be getting properties off a SettingsManager object from Emscripten code, and we shouldn't be using deprecated variables internally, strict mode or no.

File: tools/shared.py [hunk 2]
```diff
@@ -1204,6 +1208,8 @@ def __setattr__(self, attr, value):
           exit_with_error('Invalid command line option -s ' + attr + '=' + str(value) + ': ' + error_message)
         else:
           logger.debug('Option -s ' + attr + '=' + str(value) + ' has been removed from the codebase. (' + error_message + ')')
+        if attr in SettingsManager.renamed_settings:
```
Comment (jgravelle-google) [inline tools/shared.py]: This depends on all renamed settings being a subset of legacy settings, which is handled in settings.js. The data here has an unmodeled dependency, which makes me a bit uneasy.

File: tools/shared.py (hunk 1 above)
Comment (sbc100) [reply tools/shared.py]: I removed this.. and added a test.

**Comment by sbc100** [comment]
@kripken  can you take a look?

**Comment by sbc100** [comment]
This change does two things:
1. It renames BINARYEN_ASYNC_COMPILATION to WASM_ASYNC_COMPILATION
2. It add support for renamed settings where the old name is silently accept in addition to the new name. 
 However STRICT mode the old name is *not* accepted.

**Pull request merged**